pkg/services/hivelocity/device/device.go

```diff
@@ -107,6 +107,10 @@ func (s *Service) actionAssociateDevice(ctx context.Context) actionResult {
 	// list all devices
 	devices, err := s.scope.HVClient.ListDevices(ctx)
 	if err != nil {
+		if errors.Is(err, hvclient.ErrRateLimitExceeded) {
+			conditions.MarkTrue(s.scope.HivelocityMachine, infrav1.RateLimitExceeded)
+			record.Event(s.scope.HivelocityMachine, "RateLimitExceeded", "exceeded rate limit with calling Hivelocity function ListDevices")
+		}
```

Comment:
The same lines get used many times.

I would create a method `setErrorCondition(err)` and move these lines into the new method.

Reply:
I created a new function for it.